pkg/util/kernel/download_headers.go

```diff
@@ -0,0 +1,125 @@
+// +build linux
+
+package kernel
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	"github.com/DataDog/nikos/apt"
+	"github.com/DataDog/nikos/cos"
+	"github.com/DataDog/nikos/rpm"
+	"github.com/DataDog/nikos/types"
+	"github.com/DataDog/nikos/wsl"
+)
+
+// customLogger is a wrapper around our logging utility which allows nikos to use our logging functions
+type customLogger struct{}
+
+func (c customLogger) Debug(args ...interface{})                 { log.Debug(args...) }
+func (c customLogger) Info(args ...interface{})                  { log.Info(args...) }
+func (c customLogger) Warn(args ...interface{})                  { log.Warn(args...) }
+func (c customLogger) Error(args ...interface{})                 { log.Error(args...) }
+func (c customLogger) Debugf(format string, args ...interface{}) { log.Debugf(format, args...) }
+func (c customLogger) Infof(format string, args ...interface{})  { log.Infof(format, args...) }
+func (c customLogger) Warnf(format string, args ...interface{})  { log.Warnf(format, args...) }
+func (c customLogger) Errorf(format string, args ...interface{}) { log.Errorf(format, args...) }
+
+var _ types.Logger = customLogger{}
+
+// KernelHeadersDownloadDir is the folder where we'll attempt to download kernel headers
+const KernelHeadersDownloadDir = "/tmp/kernel-headers"
```

Comment:
This probably needs to be configurable, and default to something within `/var/tmp` since that lives longer than `/tmp`.